### PR TITLE
Add Contentful extension

### PR DESCRIPTION
**Pull request by ewgenius**
## Description

Extension for Contentful CMS - https://www.contentful.com/

- allows to navigate and search content library with filtering by content type and full text search by query string (all available content types fetched via api)
- view selected content details
- quickly open selected content in browser

All functionality based on official content management api - https://www.contentful.com/developers/docs/references/content-management-api/ and requires to specify Contentful space id and personal content management token in extension settings

At the moment extension provides readonly functionality, but I'm planning to add additional commands and actions in future to create and modify content, like:
- Add Entry
- Publish / Unpublish / Archive / Delete actions

## Screencast

1. create new account at https://www.contentful.com/
2. prepare some content types and content
3. create personal content management token for extension

https://user-images.githubusercontent.com/827338/209407859-7e7dced7-dc13-444b-8ce2-0fe2438f152a.mp4

Contentful data for reference:

| | |
|-|-|
|CleanShot 2022-12-24 at 01 42 40@2x|CleanShot 2022-12-24 at 01 42 44@2x|

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by igorlukanin** [comment]
I'm very much interested in this extension!

File: extensions/contentful/package.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  "author": "ewgenius",
+  "categories": [
+    "Data",
+    "Productivity"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "search",
+      "title": "Search Content",
```
Comment (pernielsentikaer) [inline extensions/contentful/package.json:16]:
```suggestion
      "title": "Search Content",
      "subtitle": "Contentful",
```

File: extensions/contentful/package.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      "description": "Search Contentful content library",
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "space",
+      "title": "Contentful space id",
```
Comment (pernielsentikaer) [inline extensions/contentful/package.json:24]:
```suggestion
      "title": "Contentful Space ID",
```

File: extensions/contentful/package.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      "description": "Contentful space id",
+      "type": "textfield",
+      "required": true
+    },
+    {
+      "name": "token",
+      "title": "Contentful management token",
```
Comment (pernielsentikaer) [inline extensions/contentful/package.json:31]:
```suggestion
      "title": "Contentful Management Token",
```

File: extensions/contentful/package.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      "description": "Contentful management token",
+      "type": "password",
+      "required": true
+    },
+    {
+      "name": "environment",
+      "title": "Contentful environment",
```
Comment (pernielsentikaer) [inline extensions/contentful/package.json:38]:
```suggestion
      "title": "Contentful Environment",
```

File: extensions/contentful/src/search.tsx [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+
+  return (
+    <List
+      isLoading={isLoading}
+      searchText={searchText}
+      onSearchTextChange={setSearchText}
+      throttle
+      isShowingDetail={showDetail}
+      searchBarAccessory={
+        <List.Dropdown tooltip="content type" isLoading={contentTypesLoading} onChange={setContentType}>
```
Comment (pernielsentikaer) [inline extensions/contentful/src/search.tsx:141]:
```suggestion
        <List.Dropdown tooltip="Content Type" isLoading={contentTypesLoading} onChange={setContentType}>
```

File: extensions/contentful/src/search.tsx [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+
+  return (
+    <List.Item
+      icon={Icon.Document}
+      key={entry.sys.id}
+      title={title}
+      actions={
+        <ActionPanel>
+          <Action.OpenInBrowser url={entryUrl} />
+          <Action title={showDetail ? "Hide Detail" : "Show Detail"} onAction={toggleDetail} />
```
Comment (pernielsentikaer) [inline extensions/contentful/src/search.tsx:203]:
```suggestion
          <Action title={showDetail ? "Hide Detail" : "Show Detail"} icon={Icon.AppWindowSidebarLeft} onAction={toggleDetail} />
```

File: extensions/contentful/src/search.tsx [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+      metadata={
+        <List.Item.Detail.Metadata>
+          <List.Item.Detail.Metadata.Label
+            title="id"
+            text={{
+              value: entry.sys.id,
+              color: Color.SecondaryText,
+            }}
+          />
+          <List.Item.Detail.Metadata.TagList title="type">
```
Comment (pernielsentikaer) [inline extensions/contentful/src/search.tsx:246]:
```suggestion
          <List.Item.Detail.Metadata.TagList title="Type">
```

File: extensions/contentful/src/search.tsx [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+            <List.Item.Detail.Metadata.TagList.Item text={type} />
+          </List.Item.Detail.Metadata.TagList>
+          <List.Item.Detail.Metadata.TagList title="status">
```
Comment (pernielsentikaer) [inline extensions/contentful/src/search.tsx:249]:
```suggestion
          <List.Item.Detail.Metadata.TagList title="Status">
```

File: extensions/contentful/src/search.tsx [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+            <List.Item.Detail.Metadata.TagList.Item text={status} color={statusColor} />
+          </List.Item.Detail.Metadata.TagList>
+          <List.Item.Detail.Metadata.Separator />
+
+          <List.Item.Detail.Metadata.Label title="Fields:" />
```
Comment (pernielsentikaer) [inline extensions/contentful/src/search.tsx:254]:
```suggestion
          <List.Item.Detail.Metadata.Label title="Fields" />
```
To keep the consistence

File: extensions/contentful/src/search.tsx [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+                key={key}
+                title={key}
+                text={typeof field === "string" ? field : JSON.stringify(field)}
+              />
+            );
+          })}
+
+          <List.Item.Detail.Metadata.Separator />
+
+          <List.Item.Detail.Metadata.Label title="Sys:" />
```
Comment (pernielsentikaer) [inline extensions/contentful/src/search.tsx:268]:
```suggestion
          <List.Item.Detail.Metadata.Label title="Sys" />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you 😊

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by ewgenius** [comment]
@pernielsentikaer hello! thanks for еру review, fixes and suggestions, committed all of them 👍

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by ewgenius** [comment]
Thank you!

**Pull request merged**